services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceHelixAdmin.java

```diff
@@ -3918,27 +3918,29 @@ private boolean hasFatalDataValidationError(PushMonitor pushMonitor, String topi
     }
   }
 
-  private void safeDeleteRTTopic(String clusterName, String storeName) {
-    boolean rtDeletionPermitted = isRTTopicDeletionPermittedByAllControllers(clusterName, storeName);
+  private void safeDeleteRTTopic(String clusterName, Store store) {
+    String realTimeTopicName = Utils.getRealTimeTopicName(store);
```

Comment:
I think I have some questions about this. how do we handle a corner case if a store is non-hybrid, but it still have a hybrid version?

Reply:
this is checked in `isRTTopicDeletionPermittedByAllControllers`